crates/nbformat/src/v4.rs

```diff
@@ -103,35 +103,38 @@ pub struct Notebook {
     pub cells: Vec<Cell>,
 }
 
-#[derive(Clone, Deserialize, Debug)]
+#[derive(Clone, Deserialize, Serialize, Debug)]
 pub struct Metadata {
+    #[serde(skip_serializing_if = "Option::is_none")]
     pub kernelspec: Option<KernelSpec>,
```

Comment:
This is only one that I was a little worried was required, but everything is optional in here.